contracts/strategy/rysk/RyskStrategy.sol

```diff
@@ -0,0 +1,240 @@
+        external
+        view
+        virtual
+        override(IStrategy)
+        returns (bool)
+    {
+        return _getTotalShares() != 0;
+    }
+
+    /// @inheritdoc IStrategy
+    function investedAssets()
+        external
+        view
+        virtual
+        override(IStrategy)
+        returns (uint256)
+    {
+        return _sharesToUnderlying(_getTotalShares());
+    }
+
+    /// @inheritdoc IStrategy
+    function invest() external virtual override(IStrategy) onlyManager {
+        uint256 balance = underlying.balanceOf(address(this));
+
+        if (balance == 0) revert StrategyNoUnderlying();
+
+        ryskLqPool.deposit(balance);
+
+        emit StrategyInvested(balance);
+    }
+
+    /// @inheritdoc IStrategy
+    function withdrawToVault(uint256 amount)
+        external
+        virtual
+        override(IStrategy)
+        onlyManager
+    {
+        if (amount == 0) revert StrategyAmountZero();
+
+        uint256 _sharesToWithdraw = _underlyingToShares(amount);
+
+        ryskLqPool.initiateWithdraw(_sharesToWithdraw);
+
+        emit RyskWithdrawalInitiated(_sharesToWithdraw);
+    }
+
+    function completeWithdrawal() external {
+        uint256 epoch = ryskLqPool.withdrawalReceipts(address(this)).epoch;
+
+        if (epoch == 0) revert RyskNoWithdrawalInitiated();
+        if (epoch == ryskLqPool.epoch())
+            revert RyskCannotComipleteWithdrawalInSameEpoch();
+
+        uint256 sharesToWithdraw = ryskLqPool
+            .withdrawalReceipts(address(this))
+            .shares;
+
+        uint256 amountWithdrawn = ryskLqPool.completeWithdraw(sharesToWithdraw);
+
+        underlying.safeTransfer(vault, amountWithdrawn);
+
+        emit StrategyWithdrawn(amountWithdrawn);
+    }
+
+    /**
+     * Get the number of Rysk liquidity pool shares owned by the strategy.
+     *
+     * @return shares owned by the strategy
+     */
+    function _getTotalShares() internal view returns (uint256) {
+        // total shares = redeemed shares + unredeemed shares
+        return
+            ryskLqPool.balanceOf(address(this)) +
+            ryskLqPool.depositReceipts(address(this)).unredeemedShares;
+    }
+
+    /**
+     * Calculates the value of Rysk liquidity pool vault shares in underlying.
+     *
+     * @param _shares number of shares
+     *
+     * @return underlying value of shares
+     */
+    function _sharesToUnderlying(uint256 _shares)
+        internal
+        view
+        returns (uint256)
+    {
+        return
+            (_shares * ryskLqPool.epochPricePerShare(ryskLqPool.epoch())) /
+            1e18;
+    }
+
+    /**
+     * Calculates the amount of underlying in number of Rysk liquidity pool shares.
+     *
+     * @param _underlying amount of underlying
+     *
+     * @return number of shares
+     */
+    function _underlyingToShares(uint256 _underlying)
+        internal
+        view
+        returns (uint256)
+    {
+        return
+            (_underlying * 1e18) /
```

Comment:
Is this multiplication really necessary? I thought the `epochPricePerShare` would already use the same units

Reply:
yes, but we are doing division on the line below so I think it has to be multiplied first

Reply:
I'm confused, the price per share should already use the same decimal places :/ we can have a call, I may be missing something